src/atomate2/common/jobs/gruneisen.py

```diff
@@ -0,0 +1,123 @@
+from typing import TYPE_CHECKING
+
+import phonopy
+from jobflow import Response, job
+from phonopy.api_gruneisen import PhonopyGruneisen
+from pymatgen.io.phonopy import get_gruneisen_ph_bs_symm_line, get_gruneisenparameter
+from pymatgen.phonon.gruneisen import (
+    GruneisenParameter,
+    GruneisenPhononBandStructureSymmLine,
+)
+from pymatgen.phonon.plotter import GruneisenPhononBSPlotter, GruneisenPlotter
+
+from atomate2.common.schemas.gruneisen import GruneisenParameterDocument
+
+if TYPE_CHECKING:
+    from pymatgen.core.structure import Structure
+
+
+@job
+def shrink_expand_structure(structure: Structure, perc_vol: float) -> Response:
+    """
+    Create structures with expanded and reduced volumes.
+
+    structure: optimized pymatgen structure obj
+    perc_vol: percentage to shrink and expand the volume
+    """
+    plus_struct = structure.copy()
+    minus_struct = structure.copy()
+
+    plus_struct.scale_lattice(volume=structure.volume * (1 + perc_vol))
+    minus_struct.scale_lattice(volume=structure.volume * (1 - perc_vol))
+
+    return Response(output={"plus": plus_struct, "minus": minus_struct})
+
+
+@job(
+    output=GruneisenParameterDocument,
+    data=[GruneisenParameter, GruneisenPhononBandStructureSymmLine],
+)
+def compute_gruneisen_param(
+    phonopy_yaml_paths_dict: dict,
+    phonon_imaginary_modes_info: dict,
+    mesh: tuple = (20, 20, 20),
+    structure: Structure = None,
+    **plot_kwargs,
+) -> GruneisenParameterDocument:
+    """
+    Compute Gruneisen parameters from phonon runs.
+
+    Requires phonopy yaml files from ground, expanded and contracted structures
+
+    phonopy_yaml_paths_dict: optimized pymatgen structure obj
+    phonon_imaginary_modes_info: dict with bool indicating if structure
+        has imaginary modes
+    mesh: sampling mesh
+    structure: pymatgen structure object at ground state
+    plot_kwargs: kwargs for pymatgen Gruneisen plotters
+    """
+    ground = phonopy.load(
+        Path(phonopy_yaml_paths_dict["ground"]) / "ground_phonopy.yaml"
+    )
+    plus = phonopy.load(Path(phonopy_yaml_paths_dict["plus"]) / "plus_phonopy.yaml")
+    minus = phonopy.load(Path(phonopy_yaml_paths_dict["minus"]) / "minus_phonopy.yaml")
+
+    if plot_kwargs.get("units") is None:
+        plot_kwargs["units"] = "thz"
+    elif plot_kwargs.get("img_format") is None:
+        plot_kwargs["img_format"] = "pdf"
+
+    gru = PhonopyGruneisen(phonon=ground, phonon_plus=plus, phonon_minus=minus)
+    gru.set_mesh(mesh=mesh)
+    gru._mesh.write_yaml(filename="gruneisen_mesh.yaml")  # noqa: SLF001
+    gruneisen_parameter = get_gruneisenparameter(
+        gruneisen_path="gruneisen_mesh.yaml", structure=structure
+    )
+    gp_plot = GruneisenPlotter(gruneisen=gruneisen_parameter)
+    gp_plot.save_plot(
+        filename="gruneisen_mesh.pdf",
+        units=plot_kwargs.get("units"),
+        img_format=plot_kwargs.get("img_format"),
+    )
+    ground.auto_band_structure()
```

Comment:
This would mean that we just have one kpath schema available.

Reply:
Yeah, it's not yet finalized.  Just got the workflow running first , indent to add kwargs

Reply:
Great, maybe, you can reuse code from the phonon workflow!

Reply:
Have added it now

Reply:
Congrats @naik-aakash . Really great already!